### PR TITLE
docs: use correct types for node in "Explicity node types" documentation

### PR DESCRIPTION
**Pull request by maks-rafalko**
## PR Checklist

- [x] Addresses an existing open issue: fixes https://github.com/typescript-eslint/typescript-eslint/issues/9457
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Issue is described in #9457

### CODE REVIEW

File: docs/developers/Custom_Rules.mdx [hunk 1]
```diff
@@ -233,15 +233,17 @@ In that case, it is best to add an explicit type declaration.
 This rule snippet targets name nodes of both function and interface declarations:
 
 ```ts
-import { AST_NODE_TYPES, ESLintUtils } from '@typescript-eslint/utils';
+import {
+  AST_NODE_TYPES,
+  ESLintUtils,
+  TSESTree,
+} from '@typescript-eslint/utils';
```
Comment (auvred) [inline docs/developers/Custom_Rules.mdx:240]: It looks like `AST_NODE_TYPES` is no longer used in this example. What do you think about removing it?

```suggestion
import { TSESTree, ESLintUtils } from '@typescript-eslint/utils';
```

**Review by auvred — CHANGES_REQUESTED** [review]

File: docs/developers/Custom_Rules.mdx (hunk 1 above)
Comment (maks-rafalko) [reply docs/developers/Custom_Rules.mdx]: sure. `ESLintUtils` is also not used in this snipped, removed.

**Review by JoshuaKGoldberg — APPROVED** [review]
🎇

**Review by auvred — APPROVED** [review]
Thanks! :rocket:

**Pull request merged**